src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -667,6 +709,20 @@ internal override void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArgu
             }
         }
 
+        private static bool? DecodeMaybeNullWhenOrNotNullWhenAttribute(AttributeDescription description, CSharpAttributeData attribute, DiagnosticBag diagnostics)
+        {
+            var arguments = attribute.CommonConstructorArguments;
+            if (arguments.Length == 1)
+            {
+                var arg = arguments[0];
+                if (arg.Type.SpecialType == SpecialType.System_Boolean)
```

Comment:
>arg.Type.SpecialType == SpecialType.System_Boolean [](start = 20, length = 50)

It doesn't look like this covers ```TypedConstantKind.Error``` case.

Reply:
Also, to avoid code duplication, perhaps it would be better to add TryDecodeValue API instead.

---
In reply to: [286592453](https://github.com/dotnet/roslyn/pull/35846#discussion_r286592453) [](ancestors = 286592453)

Reply:
`Type.SpecialType == None` for error types.

---
In reply to: [286592951](https://github.com/dotnet/roslyn/pull/35846#discussion_r286592951) [](ancestors = 286592951,286592453)

Reply:
> Type.SpecialType == None for error types.

I am talking about ```if (_kind == TypedConstantKind.Error)``` check in TryDecodeValue implementation. I am as a reviewer have no idea how it correlates with Type.SpecialType right now and whether it (the correlation) will stay in place in the future 

---
In reply to: [286606270](https://github.com/dotnet/roslyn/pull/35846#discussion_r286606270) [](ancestors = 286606270,286592951,286592453)